python/foglamp/plugins/storage/postgres/backup_restore/backup_postgres.py

```diff
@@ -47,37 +56,273 @@
-        """  Retrieves backups information
+    _logger = None
+
+    _SCHEDULE_BACKUP_ON_DEMAND = "fac8dae6-d8d1-11e7-9296-cec278b6b50a"
+    _MODULE_NAME = "foglamp_backup_postgres"
+
+    _MESSAGES_LIST = {
+
+        # Information messages
+        "i000001": "Execution started.",
+        "i000002": "Execution completed.",
+        "i000003": "On demand backup successfully launched.",
+
+        # Warning / Error messages
+        "e000000": "general error",
+        "e000001": "cannot delete/purge backup file on file system - id |{0}| - file name |{1}| error details |{2}|",
+        "e000002": "cannot delete/purge backup information on the storage layer "
+                   "- id |{0}| - file name |{1}| error details |{2}|",
+        "e000003": "cannot retrieve information for the backup id |{0}|",
+        "e000004": "cannot launch on demand backup - error details |{0}|",
+    }
+    """ Messages used for Information, Warning and Error notice """
+
+    def __init__(self, _storage):
+        self._storage = _storage
+
+        if not Backup._logger:
+            Backup._logger = _LOGGER
+
+    def get_all_backups(self,
+                        limit: int,
+                        skip: int,
+                        status: [lib.BackupStatus, None],
+                        sort_order: lib.SortOrder = lib.SortOrder.DESC) -> list:
+
+        """ Returns a list of backups is returned sorted in chronological order with the most recent backup first.
+
+        Args:
+            limit: int - limit the number of backups returned to the number given
+            skip: int - skip the number of backups specified before returning backups-
+                  this, in conjunction with the limit option, allows for a paged interface to be built
+            status: lib.BackupStatus - limit the returned backups to those only with the specified status,
+                    None = retrieves information for all the backups
+            sort_order: lib.SortOrder - Defines the order used to present information, DESC by default
+
+        Returns:
+            backups_information: list/dict - Backups information, it is provided for each :
+                id: An identifier for the backup
+                date: The date the backup started
+                status: The status of the backup; one of running, failed, completed
+        Raises:
+        """
+
+        backups_information_complete = self.get_all_backups_complete(limit, skip, status, sort_order)
+
+        backups_information = []
+        for row in backups_information_complete:
+
+            status_decoded = lib.BackupStatus.text[int(row['status'])]
+
+            backups_information.append({'id': row['id'],
+                                        'date': row['ts'],
+                                        'status': status_decoded}
+                                       )
+
+        return backups_information
+
+    def get_all_backups_complete(
```

Comment:
what is the difference between this function and get_all_backups? and is it part of the specs? If the purpose is to get the list of complete backup then it can be achieved using a simple where status=complete instead of having a different function.

Reply:
The Backup class is used for the API integration but also by the BackupProcess class to execute the actual backup.

the get_all_backups returns the information described in the spec
and the get_all_backups_complete return all the information available, for example also the 
file_name, used by the BackupProcess.

get_all_backups actually uses the get_all_backups_complete  trimming the information as required.